x-pack/test/common/utils/server_route_repository/create_supertest_service_from_repository.ts

```diff
@@ -44,49 +44,15 @@ type RepositorySupertestReturnOf<
   }>
 >;
 
-type ScopedApiClientWithBasicAuthFactory<TServerRouteRepository extends ServerRouteRepository> = (
-  kibanaServer: UrlObject,
-  username: string
-) => RepositorySupertestClient<TServerRouteRepository>;
-
-type ApiClientFromSupertestFactory<TServerRouteRepository extends ServerRouteRepository> = (
-  st: supertest.Agent
-) => RepositorySupertestClient<TServerRouteRepository>;
-
-interface RepositorySupertestClientFactory<TServerRouteRepository extends ServerRouteRepository> {
-  getScopedApiClientWithBasicAuth: ScopedApiClientWithBasicAuthFactory<TServerRouteRepository>;
-  getApiClientFromSupertest: ApiClientFromSupertestFactory<TServerRouteRepository>;
-}
-
-export function createSupertestClientFactoryFromRepository<
+export async function getApiClientFromSupertest<
   TServerRouteRepository extends ServerRouteRepository
->(): RepositorySupertestClientFactory<TServerRouteRepository> {
-  return {
-    getScopedApiClientWithBasicAuth: (kibanaServer, username) => {
-      return getScopedApiClientWithBasicAuth<TServerRouteRepository>(kibanaServer, username);
-    },
-    getApiClientFromSupertest: (st) => {
-      return getApiClientFromSupertest<TServerRouteRepository>(st);
-    },
-  };
-}
-
-function getScopedApiClientWithBasicAuth<TServerRouteRepository extends ServerRouteRepository>(
-  kibanaServer: UrlObject,
-  username: string
-): RepositorySupertestClient<TServerRouteRepository> {
-  const { password } = kbnTestConfig.getUrlParts();
-  const baseUrlWithAuth = format({
-    ...kibanaServer,
-    auth: `${username}:${password}`,
+>(
+  st: ReturnType<typeof RoleScopedSupertestProvider>
+): Promise<RepositorySupertestClient<TServerRouteRepository>> {
+  const supertestAdmin = await st.getSupertestWithRoleScope('admin', {
+    useCookieHeader: true,
```

Comment:
question: will `supertestAdmin` be used to test for private APIs? If so, all good.

But if the APIs in tests are public you need to remove this line and then API key will be generated for `admin` role and used for API calls.

Reply:
I wonder if we should rename this function to indicate that it returns an admin level client? 
Else I don't see how we'll remember to modify this behaviour when we split the tests apart.